### PR TITLE
first iteration Store class

### LINKED ISSUE #16: Store class Task #1

### PR DESCRIPTION
**Pull request by lisie-sprawy**
please check my code, i'm going in the right direction?

### CODE REVIEW

File: shm/Player.cpp [hunk 1]
```diff
@@ -21,3 +21,14 @@ size_t Player::countAvailableSpace() {
...
 }
+
+void Player::buyCargo(Cargo* cargo, size_t amount, size_t price) {
+    ship_->load(cargo);
+    money_ = money_ - price;
+    avaliableSpace_ = avaliableSpace_ - amount;
+}
+void Player::sellCargo(Cargo* cargo, size_t amount, size_t price) {
+    ship_->unload(cargo);
+    oney_ = money_ + price;
```
Comment (ArturGubala) [inline shm/Player.cpp]: You missed one letter.
Why don't you write `money_ -= price`? It's simpler and everyone knows what does it mean.

Comment (J-Cieplinski) [reply shm/Player.cpp]: +1

File: shm/Player.cpp [hunk 2]
```diff
@@ -21,3 +21,14 @@ size_t Player::countAvailableSpace() {
...
+    avaliableSpace_ = avaliableSpace_ + amount;
```
Comment (J-Cieplinski) [inline shm/Player.cpp]: Same here. Could be `avaliableSpace_ += amount`

File: shm/Player.cpp (hunk 1 above)
Comment (J-Cieplinski) [reply shm/Player.cpp]: And regarding missing letter. Compile your code before push :)

File: shm/Player.cpp (hunk 2 above)
Comment (J-Cieplinski) [reply shm/Player.cpp]: same thing in `buyCargo` class

File: shm/Store.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    if (player -> getAvaliableSpace() < amount){
+        return Response::lack_of_space;
+    }
+    player->buyCargo(cargo, amount, price);
+    return Response::done;
+}
+
+Response Store::sell(Cargo* cargo, size_t amount, Player* player){
+    const size_t price = amount * cargo->getBasePrice();
```
Comment (J-Cieplinski) [inline shm/Store.cpp]: I don't think we would want to sell cargo for base price. It makes item rarity, alcohol strength obsolete

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

File: shm/Player.cpp (hunk 1 above)
Comment (lisie-sprawy) [reply shm/Player.cpp]: Hmmm... I would give my hand to chop off that the code was compiling and there was everything. I wouldn't have a hand now. My mistake -.-'

File: shm/Player.cpp (hunk 2 above)
Comment (lisie-sprawy) [reply shm/Player.cpp]: ofc, I forgot to shorten:)

File: shm/Store.cpp (hunk 3 above)
Comment (lisie-sprawy) [reply shm/Store.cpp]: +1, right.

**Review by lisie-sprawy — COMMENTED** [review]
Thank you very much, I'll fix this issues.

**Comment by lisie-sprawy** [comment]
After merging the missing function in Ship I'll keep working. For now the code doesn't compile

File: shm/Store.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include "Store.hpp"
+
+Response Store::buy(Cargo* cargo, size_t amount, Player* player){
```
Comment (niemarcin) [inline shm/Store.cpp]: Code format issue - Check this file as you have `){` multiple times and it would be better with `) {`

File: shm/Store.hpp [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    done,
+    lack_of_money,
+    lack_of_cargo,
+    lack_of_space
+};
+
+class Store{
+public:
+    Response buy(Cargo* cargo, size_t amount, Player* player);
+    Response sell(Cargo* cargo, size_t amount, Player* player);
```
Comment (niemarcin) [inline shm/Store.hpp]: I wonder if we should provide `Cargo* cargo` and `size_t amount` or maybe `std::string name` and `size_t amount` ?

`Cargo` object has it's name and amount inside as protected fields, but we still have `amount` provided here separately as an argument to buy and sell functions. 

You don't have to implement it already - it's just a suggestions and maybe we should discuss it on discord?

File: shm/Store.cpp (hunk 4 above)
Comment (lisie-sprawy) [reply shm/Store.cpp]: ok, fixed

File: shm/Store.cpp [hunk 6]
```diff
@@ -19,7 +19,7 @@ Response Store::buy(std::shared_ptr<Cargo> cargo, size_t amount, Player* player)
 
 Response Store::sell(std::shared_ptr<Cargo> cargo, size_t amount, Player* player) {
     const size_t price = amount * cargo->getPrice();
-    *cargo -= amount;
+    *cargo += amount;
```
Comment (J-Cieplinski) [inline shm/Store.cpp]: wrong place :) when we sell we need to decrease amount of cargo. When we buy we need to increase it

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

**Review by ArturGubala — APPROVED** [review]

**Review by J-Cieplinski — APPROVED** [review]

**Review by iprice73 — APPROVED** [review]

**Pull request merged**